### PR TITLE
+doc #726 enable algolia docs search

### LINKED ISSUE #726: (re) enable Algolia docs search for Akka HTTP
Via Maxime from Algolia:

You're now a few steps away from having it working on your website:
- Copy the following CSS/JS snippets and add them to your page

```

<!-- at the end of the HEAD -->
<link rel="stylesheet" href="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css" />
<!-- at the end of the BODY -->
<script type="text/javascript" src="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js"></script>
<script type="text/javascript"> docsearch({
apiKey: '0ccbb8bf5148554a406fbf07df0a93b9',
indexName: 'akka-http',
inputSelector: '### REPLACE ME ####',
algoliaOptions: { 'facetFilters': ["version:$VERSION", "language:$LANGUAGE"] },
debug: false // Set debug to true if you want to inspect the dropdown
});
</script>
```

- Add a search input if you don't have any yet, and update the inputSelector value in the code snippet to a CSS selector that targets your input field.

- Replace $VERSION with the version you want to search on.
The list of possible version is automatically fetched from your website.
So as of today you have: 10.0.0, 10.0.1, 3.0.0-RC1, current

- Replace $LANGUAGE with the language you want to search on.
The list of possible language is harcoded in the config.
So as of today you have: java, scala

For example if you want the search on the version "10.0.0" and the language "java" just specify: 'facetFilters': ["version:10.0.0", "language:java"]

- Optionally customize the look and feel by following the DocSearch documentation (https://community.algolia.com/docsearch/documentation/)
- You can also check your configuration in the github repo (https://github.com/algolia/docsearch-configs/blob/master/configs/akka-http.json). Feel free to open pull requests!
```

### PR DESCRIPTION
**Pull request by ktoso**
Docs search is back!

Resolves #726

### CODE REVIEW

**Comment by ktoso** [comment]
Preview:

http_model_ _akka_http

**Comment by ktoso** [comment]
Resolves https://github.com/akka/akka-http/issues/726

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Failure is the known validate-pull-request issue, addressed here: https://github.com/akka/akka-http/pull/733

File: docs/src/test/java/docs/http/javadsl/server/FileUploadExamplesTest.java [hunk 1]
```diff
@@ -36,13 +36,11 @@ public void compileOnlySpec() throws Exception {
     // just making sure for it to be really compiled / run even if empty
   }
 
-  //#simple-upload
   Route uploadVideo() {
+    final Materializer materializer = null;
+    return
```
Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/server/FileUploadExamplesTest.java]: Why did you need to introduce these changes?

Comment (ktoso) [inline docs/src/test/java/docs/http/javadsl/server/FileUploadExamplesTest.java]: there were paradox errors without them.

Comment (ktoso) [inline docs/src/test/java/docs/http/javadsl/server/FileUploadExamplesTest.java]: On this one specifically notice that there's 3 `//#simple-upload` which does not work.
I don't know why it didn't fail our validation, something's wrong with PR validator.

Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/server/FileUploadExamplesTest.java]: Oh! Good to know, I'll pay more attention on next doc involved tasks

**Comment by jrudolph** [comment]
Woohoo nice one.
